src/pushsource/_impl/model/ami.py

```diff
@@ -236,8 +236,10 @@ def _from_data(cls, data):
             "boot_mode": (
                 BootMode(data.get("boot_mode")) if data.get("boot_mode") else None
             ),
-            "billing_codes": AmiBillingCodes._from_data(
-                data.get("billing_codes") or {}
+            "billing_codes": (
```

Comment:
I'm not sure about this particular change here.

The code already had handling for the data being missing.  It would give you a default-constructed `AmiBillingCodes`, which means an AmiBillingCodes object having `name=None` and `codes=[]`.

Is there a reason to change it to instead start generating `None`?  That'd be backwards-incompatible to any code expecting the former behavior.

Reply:
For some reason if this is missing in the data it throws an error. It seems that optional only wants either an AmiBillingCodes or None. I could change it so that the parameters of AmiBillingCodes are optional and that would probably pass. Is there a better way to do this?

The specific error is name is missing when it goes to parse a missing AmiBillingCodes.

Reply:
OK, I see.  So the case where it called `AmiBillingCodes._from_data({})` never actually worked.
It surely wasn't intended to be that way, but the good news is it means there's no backwards-compatibility concern with this change.